crates/router/src/types/transformers.rs

```diff
@@ -230,6 +230,176 @@ impl ForeignFrom<api_enums::PaymentMethodType> for api_enums::PaymentMethod {
     }
 }
 
+impl ForeignTryFrom<(Self, api_enums::PaymentMethodType)> for api_enums::PaymentMethod {
```

Comment:
Why do you need a `ForeignTryFrom`? You use `ForeignTryFrom` for *conversions* when you want to convert from a type `T` to a type `U`. In this case you already have both `T` and `U`. You're not converting here. You're validating. A simple function for that should suffice

Reply:
We'll pick this up in another PR